platform/src/main/java/org/stellar/anchor/platform/event/EventProcessorManager.java

```diff
@@ -0,0 +1,163 @@
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import lombok.SneakyThrows;
+import org.stellar.anchor.api.event.AnchorEvent;
+import org.stellar.anchor.api.exception.AnchorException;
+import org.stellar.anchor.event.EventService;
+import org.stellar.anchor.event.EventService.EventQueue;
+import org.stellar.anchor.platform.config.CallbackApiConfig;
+import org.stellar.anchor.platform.config.ClientsConfig;
+import org.stellar.anchor.platform.config.EventProcessorConfig;
+import org.stellar.anchor.platform.utils.DaemonExecutors;
+import org.stellar.anchor.util.Log;
+
+public class EventProcessorManager {
+  public static final String CLIENT_STATUS_CALLBACK_EVENT_PROCESSOR_NAME_PREFIX =
+      "client-status-callback-";
+  public static final String CALLBACK_API_EVENT_PROCESSOR_NAME = "callback-api";
+  private final EventProcessorConfig eventProcessorConfig;
+  private final CallbackApiConfig callbackApiConfig;
+  private final ClientsConfig clientsConfig;
+  private final EventService eventService;
+
+  private final List<EventProcessor> processors = new ArrayList<>();
+
+  public EventProcessorManager(
+      EventProcessorConfig eventProcessorConfig,
+      CallbackApiConfig callbackApiConfig,
+      ClientsConfig clientsConfig,
+      EventService eventService) {
+    this.eventProcessorConfig = eventProcessorConfig;
+    this.callbackApiConfig = callbackApiConfig;
+    this.clientsConfig = clientsConfig;
+    this.eventService = eventService;
+  }
+
+  @PostConstruct
+  public void start() {
+    if (eventProcessorConfig.getCallbackApiRequest().isEnabled()) {
+      // Create a processor for the callback API handler
+      processors.add(
+          new EventProcessor(
+              CALLBACK_API_EVENT_PROCESSOR_NAME,
+              EventQueue.TRANSACTION,
+              new CallbackApiHandler(callbackApiConfig)));
+      // Create a processor of the client status callback handler for each client defined in the
+      // clientsConfig
+      if (eventProcessorConfig.getClientStatusCallback().isEnabled()) {
+        clientsConfig
+            .getClients()
+            .forEach(
+                (clientConfig) -> {
+                  String processorName = null;
+                  switch (clientConfig.getType()) {
+                    case CUSTODIAL:
+                      processorName =
+                          CLIENT_STATUS_CALLBACK_EVENT_PROCESSOR_NAME_PREFIX
+                              + clientConfig.getSigningKey();
+                      break;
+                    case NONCUSTODIAL:
+                      processorName =
+                          CLIENT_STATUS_CALLBACK_EVENT_PROCESSOR_NAME_PREFIX
+                              + clientConfig.getDomain();
+                      break;
+                    default:
+                      errorF("Unknown client type: {}", clientConfig.getType());
```

Comment:
Should we be throwing an exception here?

Reply:
Can't really throw in lambda unless changing to `for` loop. Since this normally do not happen because it's an `enum` and relatively stable, the `errorF` should give strong warning as well.

Considering the pros and cons AGAIN, I will change to for-loop with throwing exceptions.